main.go

```
@@ -31,22 +33,45 @@ func main() {
+    r.LoadHTMLGlob("dist/*.html")
     
+    // Catch-all route to serve React app
+    r.NoRoute(func(c *gin.Context) {
+        c.File("dist/index.html")
+    })
 
-+    if err := r.Run("0.0.0.0:8080"); err != nil {
-+        log.Fatal("Failed to start server:", err)
-+    }
+    log.Println("Server starting on :8080")
+    r.Run("0.0.0.0:8080")
```

golangci-lint (1.64.8)

Error return value of `r.Run` is not checked

(errcheck)